ocean_provider/requests.py

```diff
@@ -0,0 +1,181 @@
+from eth_utils import add_0x_prefix
+from flask import request as flask_request
+from flask_sieve import JsonRequest
+from flask_sieve.rules_processor import RulesProcessor
+from flask_sieve.validator import Validator
+from ocean_utils.did import did_to_id
+
+from ocean_provider.access_token import AccessToken
+from ocean_provider.exceptions import InvalidSignatureError
+from ocean_provider.user_nonce import UserNonce
+from ocean_provider.util import get_request_data
+from ocean_provider.utils.accounts import verify_signature
+from ocean_provider.utils.basics import get_config
+
+user_nonce = UserNonce(get_config().storage_path)
+user_access_token = AccessToken(get_config().storage_path)
+
+
+class CustomJsonRequest(JsonRequest):
+    def __init__(self, request=None):
+        request = request or flask_request
+        request = get_request_data(request)
+        self._validator = CustomValidator(rules=self.rules(), messages={
+            'signature.signature': 'Invalid signature provided.',
+            'transferTxId.access_token': 'There is already a token with these parameters'  # noqa
+        }, request=request)
+
+
+class CustomValidator(Validator):
+    def __init__(
+        self, rules=None, request=None, custom_handlers=None, messages=None,
+        **kwargs
+    ):
+        super(CustomValidator, self).__init__(
+            rules, request, custom_handlers, messages, **kwargs
+        )
+        self._processor = CustomRulesProcessor()
+
+
+class CustomRulesProcessor(RulesProcessor):
+    def validate_signature(self, value, params, **kwargs):
+        self._assert_params_size(size=3, params=params, rule='signature')
+        owner = self._attribute_value(params[0])
+        did = self._attribute_value(params[1])
+        rule = params[2]
+        original_msg = f'{owner}{did}' if rule == 'customer_did' else f'{did}'
+        try:
+            verify_signature(
+                owner, value, original_msg, user_nonce.get_nonce(owner)
+            )
+            return True
+        except InvalidSignatureError:
+            return False
+
+        return False
+
+    def validate_access_token(self, value, params, **kwargs):
+        self._assert_params_size(size=2, params=params, rule='access_token')
+        did = self._attribute_value(params[0])
+        if did.startswith('did:'):
+            did = add_0x_prefix(did_to_id(did))
+
+        consumer_address = self._attribute_value(params[1])
+        tx_id = value
```

Comment:
is `value` an `access_token` or a tx_id?

Reply:
a tx_id, since the validator is set on the txId field.

Reply:
where is the access token?

Reply:
we check that there is no access_token for these values (txId, consumerAddress and did). If there is one in the database, then we reject the request with 400.